### PR TITLE
feat: Process higher order reducer

### LINKED ISSUE #5530: [Will be merged with #4432]  Create process higher order reducer
Required before finishing #4432

### PR DESCRIPTION
**Pull request by marlass**
**Why we do it?**

In cart refactor we want to optimize cart refresh/reload logic and we need know if there are any requests in progress that change the cart. Only after those finish, we want to reload the cart.

This reducer will add `processesCount` property next to `loading`, `error`, `success` flags provided by `loaderReducer`. Each action initializing entity change should extend `ProcessPushAction` and actions for success/error should extend `ProcessPopAction`.

Those actions increment/decrement `processesCount` and you can use `processesCountSelector` for example to debounce reload actions.

Closes #5530

### CODE REVIEW

File: projects/core/src/state/utils/process/process.action.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    ...loaderResetMeta(entityType),
+  };
+}
+
+export class ProcessResetAction implements ProcessAction {
+  type = PROCESS_RESET_ACTION;
+  readonly meta: ProcessMeta;
+  constructor(entityType: string) {
+    this.meta = resetMeta(entityType);
+    console.log(this.meta);
```
Comment (Platonn) [inline projects/core/src/state/utils/process/process.action.ts]: 😉

File: projects/core/src/state/utils/process/process.reducer.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+export const initialProcessState: ProcessState<any> = {
+  processesCount: 0,
+};
+
+/**
+ * Higher order reducer that adds processes count
+ */
+export function processReducer<T>(
+  processActionType: string,
```
Comment (Platonn) [inline projects/core/src/state/utils/process/process.reducer.ts]: Isn't it simply `entityType`? (analogical concern for `loaderReducer` file)

File: projects/core/src/state/utils/process/process.reducer.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+): (state: ProcessState<T>, action: ProcessAction) => ProcessState<T> {
+  return (
+    state: ProcessState<T> = {
+      ...initialProcessState,
+      ...initialLoaderState,
+    },
+    action: ProcessAction
+  ): ProcessState<T> => {
+    if (action.meta && action.meta.entityType === processActionType) {
+      const processesCount = action.meta.processesCount;
```
Comment (Platonn) [inline projects/core/src/state/utils/process/process.reducer.ts]: `processesCount` vs. `processesCountDiff`?

File: projects/core/src/state/utils/processes-loader/processes-loader.action.spec.ts [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+        const meta = popMeta(TEST_ENTITY_TYPE);
+        expect(meta).toEqual({
+          entityType: TEST_ENTITY_TYPE,
+          loader: undefined,
+          processesCountDiff: -1,
+        });
+      });
+    });
+
+    describe('resetMeta', () => {
```
Comment (Platonn) [inline projects/core/src/state/utils/processes-loader/processes-loader.action.spec.ts]: processesResetMeta ;)

**Comment by marlass** [comment]
Merged with cart refactor

**Pull request closed without merging**